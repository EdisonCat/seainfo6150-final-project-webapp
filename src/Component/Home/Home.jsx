import React from "react";
import CoverCard from "../CoverCard/CoverCard";
import styles from "./Home.module.css";
import home from "../../images/home.png";
import { Link } from "react-router-dom";

const Home = () => {
	const shows = [
		{
			title: "The Falcon and The Winter Soldier",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/falcws_lob_crd_03.jpg",
			url: "https://www.youtube.com/embed/IWBsDaFWyTE",
		},
		{
			title: "Marvel Studios: Assembled",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/assembled_lob_crd-01.jpg",
			url: "https://www.youtube.com/embed/cGdn6Fqv8j0",
		},
		{
			title: "Marvel's Behind the Mask",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/behindthemask_lob_crd_02.jpg",
			url: "https://www.youtube.com/embed/ZZKzPblr1lc",
		},
		{
			title: "WandaVision",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/wandavision_lob_crd_06.jpg",
			url: "https://www.youtube.com/embed/sj9J2ecsSpo",
		},
		{
			title: "Marvel Studios: Legends",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/legends_lob_crd_01.jpg",
			url: "https://www.youtube.com/embed/lWQF6F81kbY",
		},
		{
			title: "Marvel's 616",
			thumbnail:
				"https://terrigen-cdn-dev.marvel.com/content/prod/1x/marvel616_lob_crd_02_1.jpg",
			url: "https://www.youtube.com/embed/1ODhWjCxydU",
		},
	];
	return (
		<div className={styles.container}>
			<h1>Age of Marvel TV shows</h1>
			<div className={styles.main}>
				<img src={home} alt="mainpic" />
			</div>
			<div className={styles.list}>
				{shows.map((show, idx) => (
					<CoverCard key={idx} show={show} />
				))}
			</div>
			<br />
			<h1>
				<Link to="/signup" className={styles.link}>
					Sign Up
				</Link>{" "}
				for More Information
			</h1>
		</div>
	);
};

export default Home;