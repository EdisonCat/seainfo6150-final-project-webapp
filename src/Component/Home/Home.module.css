@import url('https://fonts.googleapis.com/css2?family=Zen+Dots&display=swap');
.container {
	/* background: linear-gradient(to right, red, darkred); */
	background: #333;
	color: white;
	display: flex;
	flex-direction: column;
	align-items: center;
	min-height: 900px;
	border-top: 1px solid gray;
	font-family: 'Zen Dots', cursive;
}
.list {
	display: flex;
	border: 1px gray solid;
	overflow: scroll;
	width: 80%;
    justify-content: space-between;
}
.link {
	color: white;
	outline: none;
	text-decoration: underline;
	text-decoration-color: #333;
	transition: 0.5s;
}
.link:hover {
	text-decoration-color: red;
}
@media screen and (max-width: 1200px) {
    .main {
        display: none;
    }
}