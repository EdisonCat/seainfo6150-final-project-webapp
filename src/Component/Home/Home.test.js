import React from "react";
import Home from "./Home.jsx";

it("renders the home component correctly", () => {
	// eslint-disable-next-line no-undef
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
