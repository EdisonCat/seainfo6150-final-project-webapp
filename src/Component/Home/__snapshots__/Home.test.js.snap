// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Age of Marvel TV shows
    </h1>
    <div
      class="main"
    >
      <img
        alt="mainpic"
        src="home.png"
      />
    </div>
    <div
      class="list"
    >
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/falcws_lob_crd_03.jpg"
        />
        <div
          class="title"
        >
          The Falcon and The Winter Soldier
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/assembled_lob_crd-01.jpg"
        />
        <div
          class="title"
        >
          Marvel Studios: Assembled
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/behindthemask_lob_crd_02.jpg"
        />
        <div
          class="title"
        >
          Marvel's Behind the Mask
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/wandavision_lob_crd_06.jpg"
        />
        <div
          class="title"
        >
          WandaVision
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/legends_lob_crd_01.jpg"
        />
        <div
          class="title"
        >
          Marvel Studios: Legends
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="thumbnail"
          class="thumbnail"
          src="https://terrigen-cdn-dev.marvel.com/content/prod/1x/marvel616_lob_crd_02_1.jpg"
        />
        <div
          class="title"
        >
          Marvel's 616
        </div>
      </div>
    </div>
    <br />
    <h1>
      <a
        class="link"
        href="/signup"
      >
        Sign Up
      </a>
       
      for More Information
    </h1>
  </div>
</div>
`;
